File: 프로그래머스/2/42885. 구명보트/구명보트.py
# 매 순간 가장 무거운 사람, 가장 가벼운 사람의 조합

def solution(people, limit):
    answer = 0
    people.sort()
    
    start = 0
    end = len(people) - 1
    
    while (start <= end):
        if people[start] + people[end] <= limit:
            start += 1
            end -= 1
            answer += 1
        else:
            end -=1
            answer += 1

    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.js
// TC: O(N log N), SC: O(1)

function solution(people, limit) {
    people.sort((a, b) => a - b);
    let i = 0, j = people.length - 1;
    
    while (i < j) {
        if (people[i] + people[j] <= limit) {
            i++; // limit 안넘으면 가장 가벼운 사람 같이 태움
        }
        j--; // 제일 무거운 사람 탑승
    }

    return people.length - i;
}